google/cloud/bigtable/row.h

```diff
@@ -40,10 +40,14 @@ class Row {
 
   /// Return the row key. The returned value is not valid
   /// after this object is deleted.
-  RowKeyType const& row_key() const { return row_key_; }
+  RowKeyType const& row_key() const& { return row_key_; }
+  /// Return the row key.
+  RowKeyType row_key() && { return std::move(row_key_); }
```

Comment:
This is returning a value, but I think rvalue overloaded methods should return `T&&` not `T`. The difference is that as written, it will return a move-constructed `T`, but if the signature returns a `T&&`, it will not even move-construct a `T`, it'll simply return the rvalue reference.

Reply:
I learn everyday.